mkosi/distributions/rhel_ubi.py

```diff
@@ -0,0 +1,153 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import urllib.parse
+from collections.abc import Sequence
+
+from mkosi.architecture import Architecture
+from mkosi.config import MkosiConfig
+from mkosi.distributions import Distribution, DistributionInstaller, PackageType
+from mkosi.distributions.centos import move_rpm_db
+from mkosi.installer.dnf import Repo, invoke_dnf, setup_dnf
+from mkosi.log import die
+from mkosi.state import MkosiState
+
+
+def join_mirror(config: MkosiConfig, link: str) -> str:
+    mirror = config.mirror or "https://cdn-ubi.redhat.com/content/public/ubi/dist/"
+    return urllib.parse.urljoin(mirror, link)
+
+
+class RhelUbiInstaller(DistributionInstaller):
```

Comment:
Ideally this would inherit from `CentOSInstaller`, but I guess that's awkward because the repository names are not the same.

I think we can just override the `repositories()` method in this class as well and then the rest of the logic would be shared. Not exactly the cleanest but sharing the logic would be nice because we don't have to fix things in two places then.

Reply:
OK, I reworked it like that. It is quite a bit shorter. Maybe in the future, if they diverge too much, it'd be better to remove the inheritance. But at least for now, it seems better.